pkg/vm/emit/emit.go

```diff
@@ -31,12 +31,11 @@ func Opcodes(w *io.BinWriter, ops ...opcode.Opcode) {
 
 // Bool emits a bool type to the given buffer.
 func Bool(w *io.BinWriter, ok bool) {
-	if ok {
-		Opcodes(w, opcode.PUSHT)
-	} else {
-		Opcodes(w, opcode.PUSHF)
+	var opVal = opcode.PUSHF
+	if !ok {
+		opVal = opcode.PUSHT
 	}
-	Instruction(w, opcode.CONVERT, []byte{byte(stackitem.BooleanT)})
+	Opcodes(w, opVal, opcode.NOT)
```

Comment:
I am wondering whether we can do `NOT+NOT` in compiler.

Reply:
Huh, I've also looked there, can be done, but I've thought that it'd seriously complicate generic conversion code which will affect every type conversion while only benefit one. Looking at it once again I see we can emit type-specific conversion easily, so let me try doing that.

Reply:
Added this one also, unfortunately it can't be extended to integers because of neo-project/neo-vm#495.